Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -190,6 +191,21 @@ extension PhotonActionSheetProtocol {
             }
         }
 
+        let removeTopSitesPin = PhotonActionSheetItem(title: Strings.RemovePinTopsiteActionTitle, iconString: "action_unpin") { action in
+            guard let url = tab.url?.displayURL,
+                let sql = self.profile.history as? SQLiteHistory else { return }
+            let absoluteString = url.absoluteString
+
+            sql.getSitesForURLs([absoluteString]) >>== { result in
+                guard let siteOp = result.asArray().first, let site = siteOp else {
+                    log.warning("Could not get site for \(absoluteString)")
+                    return
+                }
+
+                _ = self.profile.history.removeFromPinnedTopSites(site).value
```

Comment:
We want to avoid using `.value` This blocks until the query completes. In this case its not a big deal. But I think we can do better.

Reply:
We can instead use bind to have `getSitesForURLs` return another deferred via `removeFromPinnedTopSites` like this 

``` swift
sql.getSitesForURLs([string]).bind { val -> Success in
    guard let site = val.successValue else {
        return succeed()
    }

    return self.profile.history.removeFromPinnedTopSites(site)
}.uponQueue(.main) { _ in }
```